StoreApp.BLL/Services/ProductService.cs

```diff
@@ -58,6 +65,49 @@ public IEnumerable<ProductDTO> GetAll()
             return config.Map<IEnumerable<Product>, IEnumerable<ProductDTO>>(DataBase.Products.GetAll());
         }
 
+        public bool ValidateNewProduct(ProductDTO product)
+        {
+            var validate = DataBase.Products.Find(p => p.Name.Equals(product.Name) &&
+                                                       p.BrandId.Equals(product.BrandId) &&
+                                                       p.ProducerId.Equals(product.ProducerId));
+
+            foreach (var item in validate)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool ValidateEditProduct(ProductDTO productDTO)
+        {
+            var products = DataBase.Products.GetAll();
+
+            var product = products.FirstOrDefault(p => p.Equals(productDTO.Id));
+
+            if (product != null &&
+                product.Name == productDTO.Name &&
+                product.BrandId == productDTO.BrandId &&
+                product.ProducerId == productDTO.ProducerId)
+            {
+                products.ToList().ForEach(p => DataBase.Products.Detach(p));
+                return true;
+            }
+
+            product = products.FirstOrDefault(p => p.Name.Equals(productDTO.Name) &&
+                                                   p.BrandId.Equals(productDTO.BrandId) &&
+                                                   p.ProducerId.Equals(productDTO.ProducerId));
+
+            products.ToList().ForEach(p => DataBase.Products.Detach(p));
+
+            if (product == null)
+            {
+                return true;
+            }
+
+            return false;
```

Comment:
You can apply a simpler code than this one

Reply:
fixed

Reply:
fixed